test/data-driven/text-fragments-data-driven-test.js

```diff
@@ -0,0 +1,239 @@
+   * |outputDirectory|, or |inputExtension| must be different to
+   * |outputExtension|
+   * @param {String} testSuiteName - Name of the Jasmine test suite that
+   *     contains the tests run by this instance. e.g: 'My Beautiful Feature
+   * Data Driven Tests'
+   * @param {String} inputDirectory - Path to directory containing input files.
+   *     Relative to your Karma base path. e.g:
+   * 'test/data-driven/input/test-params/'
+   *
+   * Each input file must be a {@see TextFragmentDataDrivenTestInput} json
+   * serialized.
+   * @param {String} inputExtension - File extension of input files. e.g: 'json'
+   * @param {String} outputDirectory - Path to directory containing output
+   *     files. Relative to your Karma base path. e.g:
+   * 'test/data-driven/output/'
+   * @param {String} outputExtension - File extension of output files. e.g:
+   *     'out'
+   * @param {String} htmlDirectory - Path to directory containing the test
+   *     case's html pages. Relative to your Karma base path. e.g:
+   * 'test/data-driven/input/html/'
+   */
+  constructor(
+      testSuiteName, inputDirectory, inputExtension, outputDirectory,
+      outputExtension, htmlDirectory) {
+    super(
+        testSuiteName, inputDirectory, inputExtension, outputDirectory,
+        outputExtension);
+    this.htmlDirectory = htmlDirectory
+  }
+
+  /**
+   * Runs a Text Fragment data-driven test case.
+   *
+   * This method takes a test case definition in |inputText|, tries to generate
+   * and highlight a text fragment, checking if the highlighted text matches the
+   * test case's expected highlighted text.
+   * @param {String} inputText - Test case input. Expected to be a {@see TextFragmentDataDrivenTestInput} json serialized.
+   * @return {String} - The test results, having the following format:
+   *
+   * GENERATION: [Success | Failure]
+   * HIGHLIGHTING: [Success | Failure]
+   *
+   * The GENERATION line indicates if a fragment was successfully generated for the test case input.
+   * The HIGHLIGHTING line indicates if the highlighted text matched the test case's expected highlighted text.
+   */
+  generateResults(inputText) {
+    const input = this.parseInput(inputText);
+    document.body.innerHTML = __html__[this.getPathForHtml(input.htmlFileName)];
+
+    const generatedFragment = this.generateFragment(input);
+    const generationResult = generatedFragment ? true : false;
+
+    const highlightingResult =
+        generationResult && this.testHighlighting(generatedFragment, input);
+
+    const outputText =
+        this.generateOutput(generationResult, highlightingResult);
+    return outputText;
+  }
+
+  /**
+   * Generates a text fragment for a test case input.
+   * @param {TextFragmentDataDrivenTestInput} input - Test case input.
+   * @return {TextFragment|null} Generated text fragment or null if generation
+   *     failed.
+   */
+  generateFragment(input) {
+    const range = document.createRange();
```

Comment:
Small readability nit: maybe move this to line 127 below, just before it's used?

Reply:
done!